src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -0,0 +1,305 @@
+import { SkipBottomButton } from './skip_bottom_button';
+import {
+  IndexPatternField,
+  search,
+  ISearchSource,
+  TimeRange,
+  Query,
+  IndexPatternAttributes,
+} from '../../../../data/public';
+import { Chart } from '../angular/helpers/point_series';
+import { AppState } from '../angular/discover_state';
+import { SavedSearch } from '../../saved_searches';
+
+import { SavedObject } from '../../../../../core/types';
+import { Vis } from '../../../../visualizations/public';
+import { TopNavMenuData } from '../../../../navigation/public';
+
+export interface DiscoverLegacyProps {
+  addColumn: (column: string) => void;
+  fetch: () => void;
+  fetchCounter: number;
+  fetchError: FetchError;
+  fieldCounts: Record<string, number>;
+  histogramData: Chart;
+  hits: number;
+  indexPattern: IIndexPattern;
+  minimumVisibleRows: number;
+  onAddFilter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
+  onChangeInterval: (interval: string) => void;
+  onMoveColumn: (columns: string, newIdx: number) => void;
+  onRemoveColumn: (column: string) => void;
+  onSetColumns: (columns: string[]) => void;
+  onSkipBottomButtonClick: () => void;
+  onSort: (sort: string[][]) => void;
+  opts: {
+    savedSearch: SavedSearch;
+    config: IUiSettingsClient;
+    indexPatternList: Array<SavedObject<IndexPatternAttributes>>;
+    timefield: string;
+    sampleSize: number;
+  };
+  resetQuery: () => void;
+  resultState: string;
+  rows: Array<Record<string, unknown>>;
+  searchSource: ISearchSource;
+  setIndexPattern: (id: string) => void;
+  showSaveQuery: boolean;
+  state: AppState;
+  timefilterUpdateHandler: (ranges: { from: number; to: number }) => void;
+  timeRange?: { from: string; to: string };
+  topNavMenu: TopNavMenuData[];
+  updateQuery: (payload: { dateRange: TimeRange; query?: Query }, isUpdate?: boolean) => void;
+  updateSavedQueryId: (savedQueryId?: string) => void;
+  vis?: Vis;
+}
+
+export function DiscoverLegacy({
+  addColumn,
+  fetch,
+  fetchCounter,
+  fetchError,
+  fieldCounts,
+  histogramData,
+  hits,
+  indexPattern,
+  minimumVisibleRows,
+  onAddFilter,
+  onChangeInterval,
+  onMoveColumn,
+  onRemoveColumn,
+  onSetColumns,
+  onSkipBottomButtonClick,
+  onSort,
+  opts,
+  resetQuery,
+  resultState,
+  rows,
+  searchSource,
+  setIndexPattern,
+  showSaveQuery,
+  state,
+  timefilterUpdateHandler,
+  timeRange,
+  topNavMenu,
+  updateQuery,
+  updateSavedQueryId,
+  vis,
+}: DiscoverLegacyProps) {
+  const [isSidebarClosed, setIsSidebarClosed] = useState(false);
+  const toMoment = function (datetime: string) {
+    if (!datetime) {
+      return '';
+    }
+    return moment(datetime).format(opts.config.get('dateFormat'));
+  };
+  const { TopNavMenu } = getServices().navigation.ui;
+  const { savedSearch, indexPatternList } = opts;
+  // @ts-ignore
+  const bucketInterval = vis?.data?.aggs?.aggs[1]?.buckets?.getInterval();
```

Comment:
dear reviewer of the future, would appreciate a hint to make @ts-ignore go away here ❤️

Reply:
many thx 👍